presto-main/src/main/java/com/facebook/presto/sql/analyzer/RelationType.java

```diff
@@ -38,19 +39,28 @@
 @Immutable
 public class RelationType
 {
+    private final RelationId relationId;
```

Comment:
I understand. However, the class falls short of describing relation (an algebraic term). It contains `Field`-s, they in turn contain `originTable`, `relationAlias` - both nothing to do with _type_ of the relation, rather its identity. As I hardened identity of a `Field` (qualified names are not unique), I added explicit identity to the class enclosing set of fields.

That's for the explanation. Do you have any idea how to address the problem?
- we can rename `RelationType` to reveal it does not quite define the type of a relation yet
- we can move identity somewhere else (`Scope` or some new intermediate class between `Scope` and `RelationType`... `Relation`?)